kani-compiler/src/codegen_cprover_gotoc/overrides/hooks.rs

```diff
@@ -8,36 +8,37 @@
 //! It would be too nasty if we spread around these sort of undocumented hooks in place, so
 //! this module addresses this issue.
 
-use crate::codegen_cprover_gotoc::codegen::PropertyClass;
+use crate::codegen_cprover_gotoc::codegen::{bb_label, PropertyClass};
 use crate::codegen_cprover_gotoc::GotocCtx;
 use crate::unwrap_or_return_codegen_unimplemented_stmt;
 use cbmc::goto_program::{BuiltinFn, Expr, Location, Stmt, Type};
-use rustc_middle::mir::{BasicBlock, Place};
-use rustc_middle::ty::print::with_no_trimmed_paths;
-use rustc_middle::ty::{Instance, TyCtxt};
-use rustc_span::Span;
+use rustc_middle::ty::TyCtxt;
+use rustc_smir::rustc_internal;
+use stable_mir::mir::mono::Instance;
+use stable_mir::mir::{BasicBlockIdx, Place};
+use stable_mir::{ty::Span, CrateDef};
 use std::rc::Rc;
 use tracing::debug;
 
-pub trait GotocHook<'tcx> {
+pub trait GotocHook {
     /// if the hook applies, it means the codegen would do something special to it
-    fn hook_applies(&self, tcx: TyCtxt<'tcx>, instance: Instance<'tcx>) -> bool;
+    fn hook_applies(&self, tcx: TyCtxt, instance: Instance) -> bool;
     /// the handler for codegen
     fn handle(
         &self,
-        tcx: &mut GotocCtx<'tcx>,
-        instance: Instance<'tcx>,
+        gcx: &mut GotocCtx,
+        instance: Instance,
         fargs: Vec<Expr>,
-        assign_to: Place<'tcx>,
-        target: Option<BasicBlock>,
-        span: Option<Span>,
+        assign_to: Place,
+        target: Option<BasicBlockIdx>,
+        span: Span,
```

Comment:
Again, I'd prefer to keep `span` as `Option<Span>` until we're 100% sure we can remove it from everywhere.

Reply:
Why? We should minimize the amount of places with `Option<Span>`. See my comment above on why we need them in the first place.

All callers of this function have access to a valid Span.

Reply:
Ok, you convinced me to minimize `Option<Span>` here.